pkg/webui/console/lib/events/utils.js

```diff
@@ -23,13 +23,29 @@ export const defineSyntheticEvent = name => data => ({
   data,
 })
 
-export const createSyntheticEventFromError = error => {
+const convertError = error => {
   if (error instanceof Error) {
-    const errorString = error.toString()
-    if (error.message === 'network error' || error.message === 'Error in body stream') {
-      return createNetworkErrorEvent({ error: errorString })
+    return {
+      ...error,
+      message: error.message,
+      name: error.name,
+      // The stack is omitted intentionally, as it is not relevant for a user.
     }
+  }
+  return error
+}
 
-    return createUnknownErrorEvent({ error: errorString })
+export const createSyntheticEventFromError = error => {
```

Comment:
I did not find a good way to import the actual _types_ in order to do `instanceof` from the SDK, so I used the `name` instead. I am open to suggestions if we can do that.

Reply:
We could create all relevant error classes in JS and then create a mapping to do sth like:
```js

class ConnectionError extends Error {
  constructor(err) {
    this.name = err.name
    this.code = err.code
    this.message = err.message
    // Add any other property specific to the error type
    …
  }
}

const classMap = {
  "ConnectionError": ConnectionError,
  …
}
```

```js
// In the error ingestion the error would then be instantiated like this
catch (err) {
  if (isContractualError(err)) { // some form of check that the error is a well-known error
    const error = new classMap[err.name]
    throw error
  }
  throw new Error("An unknown error occurred")
}
```

```js
// Then in the err util it could be checked like this
if (error instanceof classMap["ConnectionError"]) {
  …
}
```

Reply:
My point was that if we do that, we don't guarantee that the type caught is really the intended type - if two files in the JS SDK (accidentally) decide to define a `ConnectionError`, and we use `name` to identify name, we may mix them up, while the type would correctly be identified via `instanceof`.

I think this is more goldplating at the moment, and if we end up with such a situation we should just create an explicit errors package in the SDK in order to avoid duplication or conflicts.